doc/isis/frr-isis-sonic-yang-model-hld.md

```diff
@@ -0,0 +1,652 @@
+                        "Set BFD to use a pre-configured profile.";
+                }
+
+                leaf metric {
+                    type uint32 {
+                        range "0..16777215";
+                    }
+                    default "0";
+                    description
+                        "The metric value of this interface.";
+                }
+
+                leaf csnp_interval {
+                    type uint16 {
+                        range "1..600";
+                    }
+                    units "seconds";
+                    default "10";
+                    description
+                        "Complete Sequence Number PDU (CSNP) generation interval.";
+                }
+
+                leaf psnp_interval {
+                    type uint16 {
+                        range "1..120";
+                    }
+                    units "seconds";
+                    default "2";
+                    description
+                        "Partial Sequence Number PDU (PSNP) generation interval.";
+                }
+
+                leaf hello_interval {
+                    type uint32 {
+                        range "1..600";
+                    }
+                    units "seconds";
+                    default "3";
+                    description
+                        "Hello interval between consecutive hello messages. Interval will depend on multiplier.";
+                }
+
+                leaf hello_multiplier {
+                    type uint16 {
+                        range "2..100";
+                    }
+                    default "10";
+                    description
+                        "Multiplier for the hello holding time.";
+                }
+
+                uses isis-authentication; 
+
+            }  // list ISIS_INTERFACE
+
+        } // container ISIS_INTERFACE
+```
+
+#### SONiC ISIS Authentication Groupings
+Authentication leafs used to define isis authentication options. 
+ 
+ 
+``` 
+    grouping isis-authentication {
+        leaf authentication_key {
+            type string {
+                length "1..254";
+            }
+            must "(../authentication_type or not(../authentication_key))" {
+                error-message
+                    "If authentication_key is specified, then authentication_type must also be specified.";
+            }
+            description
+                "Authentication password.";
+        }
+
+        leaf authentication_type {
+            type stypes:authentication-type;
+            must "(../authentication_key or not(../authentication_type))" {
+                error-message
+                    "If authentication_type is specified, then authentication_key must also be specified.";
+            }
+            description
+                "This grouping defines keychain configuration type.";
+        }
+    }
+```
+
+##### SONiC ISIS Defined Types
+Types defined in sonic-types.yang.j2.
```

Comment:
Suggest to just follow typedefs from standard openconfig model
https://github.com/openconfig/public/blob/master/release/models/isis/openconfig-isis-types.yang

Reply:
All defined ISIS types are now from openconfig isis types